### PR TITLE
Explicitly set namespace in tackle's kubectl calls

### PR DESCRIPTION
**Pull request by diversario**
If the context has a non-default namespace set (e.g., `jx` when working with Jenkins X), then `oauth-token` secret ends up in that namespace, whereas the rest of the components are created in `default` namespace. This also enables namespacing all Prow resources in the future.

For admin role binding I'm passing hardcoded `default` because it's a `clusterrolebinding` and the namespace is ignored.

### CODE REVIEW

File: prow/cmd/tackle/main.go [hunk 1]
```diff
@@ -459,8 +459,8 @@ func githubClient(tokenPath string, dry bool) (*github.Client, error) {
 	return github.NewClient(gen, "https://api.github.com"), nil
 }
 
-func applySecret(ctx, name, key, path string) error {
-	return applyCreate(ctx, "secret", "generic", name, "--from-file="+key+"="+path)
+func applySecret(ctx, ns, name, key, path string) error {
+	return applyCreate(ctx, ns, "secret", "generic", name, "--from-file="+key+"="+path)
```
Comment (fejta) [inline prow/cmd/tackle/main.go]: How about
a) adding --namespace=default into this line
b) printing an error/warning when the context sets a context
c) then after that creating a plan to make tackle and starter.yaml allow you to deploy to a non-default namespace?

Comment (diversario) [reply prow/cmd/tackle/main.go]: Do you mean 
```
return applyCreate(ctx, "secret", "generic", name, "--from-file="+key+"="+path, "--namespace=default")
```
?

Also, not sure what you mean in b)

File: prow/cmd/tackle/main.go [hunk 2]
```diff
@@ -386,8 +386,8 @@ func applyCreate(ctx string, args ...string) error {
 	return nil
 }
 
-func apply(ctx string, in io.Reader) error {
-	apply := exec.Command("kubectl", "--context="+ctx, "apply", "-f", "-")
+func apply(ctx string, ns string, in io.Reader) error {
+	apply := exec.Command("kubectl", "--context="+ctx, "--namespace="+ns, "apply", "-f", "-")
```
[the remark this answers is not in the document]
Comment (diversario) [reply prow/cmd/tackle/main.go]: Right. I was thinking about ensuring that all resources get into the same namespace and, coincidentally, namespace in the code matches namespace in the `starter.yaml`. Making this work with a non-default ns would be more work. But, to your question, the namespace var is hardcoded to `default` [here](https://github.com/kubernetes/test-infra/blob/529e733a2d65e167385860a8210b40fd42681eaa/prow/cmd/tackle/main.go#L781) so there's no disagreement with the yaml.

File: prow/cmd/tackle/main.go (hunk 1 above)
Comment (fejta) [reply prow/cmd/tackle/main.go]: Yes that's what I mean for a

For (b) I mean if I'm selecting a context that specifies a different default namespace, but our tooling forces the namespace to be default we should error or warn so the user isn't going to be surprised.

**Pull request merged**